Document the replaceUsernames setting in .drone.star (#3091)
owncloud/docs: modules/developer_manual/pages/testing/acceptance-tests.adoc

}
----

If you are running tests locally, define the environment variable `REPLACE_USERNAMES` to be true:

[source,bash]
----
export REPLACE_USERNAMES=true
----

You can also run the acceptance tests in CI with the replaced user attributes. This is useful if you want to
run many acceptance tests with an unusual combination of usernames, display names, email addresses and

> passwords without taking up many hours on a local machine. The PR is not to be merged, it is just a way to

Reviewer: ```suggestion
passwords without taking up many hours on a local machine. _A testing PR necessary to do so but is not to be merged_, it is just a way to
```

get test results. In the `acceptance` section of `.drone.star` switch on the `replaceUsernames` setting.

> Commit the changes to `.drone.star` and `tests/acceptance/usernames.json`, push to GitHub, and make a draft PR:

Reviewer: ```suggestion
Commit the changes to `.drone.star` and `tests/acceptance/usernames.json`, push to GitHub, and make a _draft_ PR:
```

> [source,bash]

Reviewer: ```suggestion
**Example Changes:**
[source,bash]
```

}
----

If you are running tests locally, define the environment variable `REPLACE_USERNAMES` to be true:

[source,bash]
----
export REPLACE_USERNAMES=true
----

You can also run the acceptance tests in CI with the replaced user attributes. This is useful if you want to
run many acceptance tests with an unusual combination of usernames, display names, email addresses and
passwords without taking up many hours on a local machine. _A PR is necessary for testing but is not to be merged_,
it is just a way to get test results. In the `acceptance` section of `.drone.star` switch on the `replaceUsernames` setting.
Commit the changes to `.drone.star` and `tests/acceptance/usernames.json`, push to GitHub, and make a _draft_ PR:

**Example Changes**
[source,bash]
----
'acceptance': {
  'api': {
    'suites': [
      'apiAuth',
      'apiAuthOcs',
      'apiAuthWebDav',
      # and so on...
    ],
    'replaceUsernames': True,
  },
},
----

> When the acceptance tests are run, the user names and attributes will be replaced.

Reviewer: ```suggestion
When the acceptance tests are run, the user names and attributes will be replaced.
When the acceptance tests have been successfully finished, do not forget to close that PR and add a comment describing the success status.
```
